src/libs/actions/Report.js

```diff
@@ -754,16 +759,15 @@ function fetchAllReports(
             // We are waiting 8 seconds since this provides a good time window to allow the UI to finish loading before
             // bogging it down with more requests and operations.
             }, shouldDelayActionsFetch ? 8000 : 0);
-
+        })
+        .finally(() => {
             // Update currentlyViewedReportID to be our first reportID from our report collection if we don't have
             // one already.
-            if (!shouldRedirectToReport || lastViewedReportID) {
-                return;
+            if (shouldRedirectToReport || !lastViewedReportID) {
+                const firstReportID = _.first(reportIDs);
+                // eslint-disable-next-line no-use-before-define
+                updateCurrentlyViewedReportID(firstReportID);
             }
-
-            const firstReportID = _.first(reportIDs);
-            const currentReportID = firstReportID ? String(firstReportID) : '';
-            Onyx.merge(ONYXKEYS.CURRENTLY_VIEWED_REPORTID, currentReportID);
```

Comment:
Moved this logic to `finally` as it's important to set the `CURRENTLY_VIEWED_REPORTID` during initialization, even in the event of an error